File: sts/grocOrder/src/main/java/demo/grocItem/GrocItem.java
package demo.grocItem;

public class GrocItem
{
    private String _itemName;
    private FoodType _itemFoodType;
    private double _itemPrice;
    
    public void set_itemName(String _itemName)
    {
        this._itemName = _itemName;
    }
    public String get_itemName()
    {
        return _itemName;
    }
    public void set_itemFoodType(FoodType _itemFoodType)
    {
        this._itemFoodType = _itemFoodType;
    }
    public FoodType get_itemFoodType()
    {
        return _itemFoodType;
    }
    public void set_itemPrice(double d)
    {
        this._itemPrice = d;
    }
    public double get_itemPrice()
    {
        return this._itemPrice;
    }
    
    
    
    public GrocItem(String name)
    {
        this.set_itemName(name);
        this.set_itemFoodType(FoodType.FOOD);
    }
    
    public GrocItem(String name, FoodTypeService fts)
    {
        this.set_itemName(name);
        this.set_itemFoodType(fts.DetermineFoodType(this.get_itemName()));
        
    }
    

}

File: GrocOrder/GrocOrder/GrocItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace GrocOrder
{
    public enum FoodType
    {
        FOOD, NONFOOD
    }
    
    public class GrocItem
    {
        public String ItemName { get; set; }
        public FoodType ItemFoodType { get; set; }

        public GrocItem(string ItemName)
        {
            this.ItemName = ItemName;
            this.ItemFoodType = FoodType.FOOD;
        }

        public GrocItem(string ItemName, FoodTypeService fts)
        {
            this.ItemName = ItemName;
            this.ItemFoodType = fts.DetermineFoodType(this.ItemName);
        }

    }
}
